### PR TITLE
Add modal editor for use in product editor

### LINKED ISSUE #37898: Iframe editor should be placed inside of a modal
As part of https://github.com/woocommerce/woocommerce/pull/37570, the `IframeEditor` was introduced.  Currently it uses a fixed position to overlay the screen, but this should instead be placed inside of a modal.

Note that this issue is simply placement inside of the modal and does not include the fixed toolbar or inspector sidebar.

### Screenshots

**Current**
Image

**Designs**

Image

### Acceptance criteria

* [ ] The `IframeEditor` should appear inside of a Modal
* [ ] The modal header title should be changeable by the consumer
* [ ] Blocks and callbacks should continue to work as expected

### PR DESCRIPTION
**Pull request by joshuatf**
### Submission Review Guidelines:

- I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
- I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
- I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/). 
- Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

Adds a modal editor that can be used in the product editor for descriptions and other fields that require separate editors.

This PR only addresses fitting the iframe editor into the editor and does not include the toolbar, sidebars, or other functionality which will be addressed in follow-ups.

Screen Shot 2023-04-21 at 1 29 02 PM

Closes #37898 .

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Navigate to Tools -> WCA Test Helper -> Features and enable the new product blocks editing experience
2. Navigate to Products -> Add new
3. Click on "Add description" under the description
4. Add some content via the blocks editor
5. Click the modal's close button ("x")
6. Make sure the description button now reads "Edit description"
7. Save the product
8. Refresh the page
9. Click "Edit description"
10. Make sure the description has persisted

### CODE REVIEW

**Comment by joelclimbsthings** [comment]
Hey @joshuatf , I attempted to review this but the pnpm upgrades were causing issues. I attempted to downgrade again, but still no 🎲 . This will need a rebase as is, so feel free to ping me for that review when that's done!

**Comment by joshuatf** [comment]
Hey @joelclimbsthings I just rebased this and should be good for review.  You may still have to upgrade pnpm or clean cache as I know these have been updated in `trunk`.

File: packages/js/product-editor/src/style.scss [hunk 1]
```diff
@@ -13,8 +13,7 @@
 
 @import 'components/radio-field/style.scss';
 @import 'components/iframe-editor/style.scss';
-@import 'components/details-categories-field/style.scss';
```
Comment (joelclimbsthings) [inline packages/js/product-editor/src/style.scss:17]: Did you intend to remove these two files for the categories field?

File: packages/js/product-editor/src/components/modal-editor/style.scss [hunk 2]
```diff
@@ -0,0 +1 @@
+@import './modal-editor.scss';
```
Comment (joelclimbsthings) [inline packages/js/product-editor/src/components/modal-editor/style.scss]: I'm curious why we have multiple levels of scss imports here? Do we need both this file and `modal-editor-scss` files to exist?

**Review by joelclimbsthings — CHANGES_REQUESTED** [review]
Nice work @joshuatf ! I had a couple minor questions regarding the styles. I'm also assuming that the appearance isn't intended to be part of this PR, but just in case I'll mention that the paragraph blocks still appear gray on my end. 

image

File: packages/js/product-editor/src/components/modal-editor/style.scss (hunk 2 above)
Comment (joshuatf) [reply packages/js/product-editor/src/components/modal-editor/style.scss]: I was following our pattern of having an `index.ts` file with a separate `component-name.tsx` file that is imported there.  It's easier to find and identify stylesheets this way that are named `component-name.scss`.  It also creates an entry file when multiple imports do exist.

That said, I'm not overly opinionated on this and have removed it for the time being.

File: packages/js/product-editor/src/style.scss (hunk 1 above)
Comment (joshuatf) [reply packages/js/product-editor/src/style.scss:17]: Nope.  Error on my part during wonky rebase.  Added back in.

**Review by joelclimbsthings — APPROVED** [review]
Thanks @joshuatf , all looks well on my end 🚢

**Pull request merged**